MekHQ/src/mekhq/campaign/Campaign.java

```diff
@@ -7906,79 +7910,86 @@ public void initTimeInService() {
-         */
-        for (Person p : getPersonnel()) {
-            Date join = null;
-            for (LogEntry e : p.getPersonnelLog()) {
-                if (e.getDesc().startsWith("Joined ")) {
-                    join = e.getDate();
-                    break;
+
+            /*
+            * Go through all the personnel records and assume the earliest date is the date
+            * the unit was founded.
+            */
+            Date founding = null;
+            for (Person p : getPersonnel()) {
+                for (LogEntry e : p.getPersonnelLog()) {
+                    if (null == founding || e.getDate().before(founding)) {
+                        founding = e.getDate();
+                    }
                 }
             }
-            if (null != join && join.equals(founding)) {
-                p.setFounder(true);
-            }
-            if (p.getPrimaryRole() == Person.T_MECHWARRIOR
-                    || (p.getPrimaryRole() == Person.T_AERO_PILOT && getCampaignOptions().getAeroRecruitsHaveUnits())
-                    || p.getPrimaryRole() == Person.T_PROTO_PILOT) {
+            /*
+            * Go through the personnel records again and assume that any person who joined
+            * the unit on the founding date is one of the founding members. Also assume
+            * that MWs assigned to a non-Assault 'Mech on the date they joined came with
+            * that 'Mech (which is a less certain assumption)
+            */
+            for (Person p : getPersonnel()) {
+                Date join = null;
                 for (LogEntry e : p.getPersonnelLog()) {
-                    if (e.getDate().equals(join) && e.getDesc().startsWith("Assigned to ")) {
-                        String mech = e.getDesc().substring(12);
-                        MechSummary ms = MechSummaryCache.getInstance().getMech(mech);
-                        if (null != ms && (p.isFounder()
-                                || ms.getWeightClass() < megamek.common.EntityWeightClass.WEIGHT_ASSAULT)) {
-                            p.setOriginalUnitWeight(ms.getWeightClass());
-                            if (ms.isClan()) {
-                                p.setOriginalUnitTech(2);
-                            } else if (ms.getYear() > 3050) {
-                                /*
-                                 * We're only guessing anyway, so we use this hack to avoid actually loading the
-                                 * entity to check for IS2
-                                 */
-                                p.setOriginalUnitTech(1);
-                            }
-                            if (null != p.getUnitId() && null != units.get(p.getUnitId())
-                                    && ms.getName().equals(units.get(p.getUnitId()).getEntity().getShortNameRaw())) {
-                                p.setOriginalUnitId(p.getUnitId());
+                    if (e.getDesc().startsWith("Joined ")) {
```

Comment:
I know there's not really a way to address this, but :(

Reply:
Definitely not for old campaigns. I'll noodle on it some.